src/lang/modifyAst/addFillet.ts

```diff
@@ -153,48 +108,82 @@ export function addFillet(
   // CallExpression - no fillet
   // PipeExpression - fillet exists
 
-  let newExtrudePipeExpression: PipeExpression
-
-  if (extrudeType === 'CallExpression') {
-    // 1. no fillet case
-    const extrudeCall = extrudeDeclarator.init as CallExpression
-    newExtrudePipeExpression = {
-      type: 'PipeExpression',
-      start: extrudeCall.start,
-      end: extrudeCall.end,
-      body: [extrudeCall, filletCall],
-      nonCodeMeta: {
-        nonCodeNodes: {},
-        start: [],
+  const getPathToNodeOfFilletLiteral = (
+    pathToExtrudeNode: PathToNode,
+    extrudeDeclarator: VariableDeclarator,
+    tag: string
+  ): PathToNode => {
+    let pathToFilletObj: any
+    let inFillet = false
+    traverse(extrudeDeclarator.init, {
+      enter(node, path) {
+        if (node.type === 'CallExpression' && node.callee.name === 'fillet') {
+          inFillet = true
+        }
+        if (inFillet && node.type === 'ObjectExpression') {
+          const hasTag = node.properties.some((prop) => {
+            const isTagProp = prop.key.name === 'tags'
+            if (isTagProp && prop.value.type === 'ArrayExpression') {
+              return prop.value.elements.some((element) => {
+                // console.log()
+                return element.type === 'Identifier' && element.name === tag
+              })
+            }
+            return false
+          })
+          if (!hasTag) return false
+          pathToFilletObj = path
+          node.properties.forEach((prop, index) => {
+            if (prop.key.name === 'radius') {
+              pathToFilletObj.push(
+                ['properties', 'ObjectExpression'],
+                [index, 'index'],
+                ['value', 'Property']
+              )
+            }
+          })
+        }
```

Comment:
While a little more verbose, this not hardcoding values for the path to node should make it more robust. I made a task on https://github.com/KittyCAD/modeling-app/issues/2606 to write a test for this util